File: mvp_rxjava_retrofit/src/main/java/com/example/mvp_rxjava_retrofit/adapter/ArticleAdapter.java
package com.example.mvp_rxjava_retrofit.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.mvp_rxjava_retrofit.R;
import com.example.mvp_rxjava_retrofit.bean.ArticleBean;

import java.util.ArrayList;


public class ArticleAdapter extends RecyclerView.Adapter<ArticleAdapter.ViewHolder> {
    private final LayoutInflater mInflater;
    private final ArrayList<ArticleBean> mList;

    public ArticleAdapter(Context context, ArrayList<ArticleBean> list) {
        this.mList = list;
        mInflater = LayoutInflater.from(context);
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = mInflater.inflate(R.layout.item_layout, parent, false);
        return new ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.tvName.setText(mList.get(position).getName());
    }

    @Override
    public int getItemCount() {
        return mList.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvName;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvName = itemView.findViewById(R.id.tv_name);
        }
    }
}

File: mvvm_coroutines_retrofit_flow_hilt/src/main/java/com/example/mvvm_coroutines_retrofit_flow_hilt/ui/article_list/adapter/ArticleAdapter.kt
package com.example.mvvm_coroutines_retrofit_flow_hilt.ui.article_list.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.mvvm_coroutines_retrofit_flow_hilt.data.bean.ArticleBean
import com.example.mvvm_coroutines_retrofit_flow_hilt.databinding.ItemLayoutBinding

class ArticleAdapter(private val mContext: Context, private val mList: ArrayList<ArticleBean>) :
    RecyclerView.Adapter<ArticleAdapter.ViewHolder>() {
    private val mLayoutInflater: LayoutInflater = LayoutInflater.from(mContext)

    class ViewHolder(binding: ItemLayoutBinding) : RecyclerView.ViewHolder(binding.root) {
        val tvName = binding.tvName
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(ItemLayoutBinding.inflate(mLayoutInflater, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val article: ArticleBean = mList[position]
        holder.tvName.text = "${article.name} - ${article.order}"
    }

    override fun getItemCount(): Int = mList.size
}